src/main/java/teammates/logic/core/StudentsLogic.java

```diff
@@ -628,20 +620,33 @@ private StudentEnrollDetails enrollStudent(StudentAttributes validStudentAttribu
         return studentList;
     }
 
-    private List<String> getInvalidityInfoInDuplicatedEmail(String email,
-            List<String> studentEmailList, String[] linesArray) {
-        List<String> info = new ArrayList<>();
-        info.add("Same email address as the student in line \"" + linesArray[studentEmailList.indexOf(email) + 1] + "\"");
-        return info;
+    private String getInvalidityInfoInDuplicatedEmail(String[] linesArray, int duplicateEmailIndex) {
+        return "Same email address as the student in line \"" + linesArray[duplicateEmailIndex + 1] + "\"";
     }
 
-    private boolean isStudentEmailDuplicated(String email,
-            List<String> studentEmailList) {
-        return studentEmailList.contains(email);
+    /**
+     * lines is the enrollment lines entered by the instructor.
+     */
+    private String invalidStudentInfo(StudentAttributes student, String line) {
```

Comment:
1. What I meant was, change the name of the variable `line` to something that is more descriptive, and update the header comments too. 

1. Also, to be more precise, this `line` isn't really the enrollment line entered by the instructor, because it has been sanitised already.

1. Follow the standard of header comments as per the one I gave you earlier, using annotations such as {@code} when denoting code, and also describing the return values.

1. You didn't specify what is this method returning.